src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2ConnectionException.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+
+namespace System.Net.Http
+{
+    [Serializable]
+    internal sealed class Http2ConnectionException : Http2ProtocolException
+    {
+        public Http2ConnectionException(Http2ProtocolErrorCode protocolError)
+            : base(SR.Format(SR.net_http_http2_connection_error, GetName(protocolError), ((int)protocolError).ToString("x")), protocolError)
+        {
+        }
+    }
```

Comment:
This needs a
```
private Http2ConnectionException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
ctor.  The current `private` one on the `Http2ProtocolException` will then need to be made `protected`.

Reply:
will do